src/ZendeskApi.Client/Extensions/HttpResponseExtensions.cs

```diff
@@ -1,47 +1,129 @@
 using System.Net;
 using System.Net.Http;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
 using ZendeskApi.Client.Exceptions;
 
 namespace ZendeskApi.Client.Extensions
 {
-    public static class HttpResponseExtensions
+    internal static class HttpResponseExtensions
     {
-        public static async Task ThrowIfUnsuccessful(
+        internal static async Task<HttpResponseMessage> IsNullWhen(
```

Comment:
Should this not be returning a bool, with the name starting with `Is`?

Reply:
Nope. If you look at the usage I want to chain together a series of actions after a request so I can compose together flows for slightly different usages: https://github.com/justeat/ZendeskApiClient/pull/195/files#diff-c0163894d98e12399628d6a8ef46031cR69

Reply:
As it then lets me reuse the functions for requests which don't quite fit the normal pattern: https://github.com/justeat/ZendeskApiClient/pull/195/files#diff-e167f1e3a33337c1008dc6d0b9419135R37

Reply:
Might still be preferable to call it `TransformNullIf()` or `MakeNullIf()`; i'd concur that `IsWhatever` generally implies a bool-typed member.

Reply:
Changed it to SetToNullWhen which represents what its doing more accurately?